ci_tools/check_slots.py

```diff
@@ -26,82 +27,133 @@
 missing_attribute_nodes = []
 missing_from_all = []
 
+error_collection = {
+    'missing_all':[],
+    'non_alphabetical_all':[],
+    'missing_slots':[],
+    'overridden_slots':[],
+    'missing_attribute_nodes':[],
+    'missing_from_all':[]
+}
+
+def extract_dict_elements(input_dict):
+    """
+    Extracts specific elements from the input dictionary based on the given keys and returns a new dictionary.
+
+    Args:
+        input_dict (dict): The input dictionary from which elements will be extracted.
+
+    Returns:
+        dict: A new dictionary containing the extracted elements.
+    """
+    output_dict = {'title':"Pyccel_lint", 'summary':"## Check Slots:\n\n",'annotations':[]}
+    for values in input_dict.items():
+        value_list = values[-1]
+        if value_list:
+            output_dict['summary'] += f"### {value_list[0]['title']} :\n\n"
+            output_dict['summary'] += "\n\n".join(f"-  {value['message']}" for value in value_list)
+            output_dict['summary'] += "\n\n"
+            output_dict['annotations'].extend(value['annotations'] for value in value_list)
+    return output_dict
+
+def fill_dictionary(title, message, file, start_line, end_line, annotation_level, annotation_msg):
+    """
+    Fills a dictionary with the provided data.
+
+    Args:
+        data (dict): The data to populate the dictionary. It should be a dictionary
+                     where the keys represent the dictionary keys, and the values
+                     represent the corresponding values.
+
+    Returns:
+        dict: A dictionary filled with the provided data.
+    """
+    index = file.find("site-packages")
+    filled_dict = {
+        'title':title,
+        'message':f"`{message}`",
+        'annotations':{
+            'path':file[index + len("site-packages") + 1:],
+            'start_line':start_line,
+            'end_line':end_line,
+            'annotation_level':annotation_level,
+            'message':annotation_msg
+        }
+    }
+    return filled_dict
+
 for mod_name in ast_modules:
     mod = importlib.import_module('pyccel.ast.'+mod_name)
     all_attr = getattr(mod, '__all__', None)
     if all_attr:
         sorted_all = list(all_attr)
         sorted_all.sort()
         if sorted_all != list(all_attr):
-            non_alphabetical_all.append(mod_name)
+            lines = inspect.getsource(mod).splitlines()
```

Comment:
Is it not possible to do `inspect.getsource(mod.__all__)`?

Reply:
I can't, I need to count the lines so I could return the start line and the end.

Reply:
Ah that's a shame. I hoped `getsource` would return multiple lines :(

Reply:
we can't do inspect.getsource(mod.__all__) because mod.__all__ is a tuple and for this case I need all the file so I can get the first line number where the __all__ attribute is inserted in the file then get the end line number where it's finished. I don't think their is a function that do so.